pkgs/servers/sql/postgresql/default.nix

```diff
@@ -14,8 +14,15 @@ let
     outputs = [ "out" "lib" "doc" "man" ];
     setOutputFlags = false; # $out retains configureFlags :-/
 
+    combinedLibXML2 = symlinkJoin {
```

Comment:
Hmm, why is this needed?

Reply:
Possibly, but I just realised that the upstream broke cross-compilation capability, possibly due to #44068 by @Ericson2314. I will close this PR for now.

Reply:
@dezgeg I tried. However, the `configure` of `postgresql` expects the `libxml2` path to contain both its headers and its static libraries.

We would either patch the `configure`, or just supply the combined `libxml2`.

Reply:
If you pass it as a `buildInput`, its dev outputs will be passed with the headers, and the `-L` to the libs added by cc-wrapper will probably leave the configure script unable to notice the difference. If you make it just a nativeBuildInput you will not get the `-L`.

Reply:
@Ericson2314 I just confirmed that if `libxml2` is added to `buildInputs`, the static library is visible to the cross compilers. However, the headers are not picked up by the `configure` script.

Sorry that I am new to this cross compiling business. Should we modify `CFLAGS` to point to the headers?

Reply:
Currently `libxml2` export headers in the `include/libxml2` folder in the Nix store, but the `CFLAGS` is pointing to `include` only. When `configure` is resolving `#include <libxml/parser.h>`, it is looking at the `include` folder only and this fails the `configure` checks.

The build works if I point the headers to `include/libxml2`. The question now is whether `libxml2` Nix expression evaluating correctly.